homeassistant/components/fully_kiosk/entity.py

```diff
@@ -54,3 +58,28 @@ def __init__(self, coordinator: FullyKioskDataUpdateCoordinator) -> None:
                 (CONNECTION_NETWORK_MAC, coordinator.data["Mac"])
             }
         self._attr_device_info = device_info
+
+    async def mqtt_subscribe(
+        self, event: str | None, event_callback: CALLBACK_TYPE
+    ) -> CALLBACK_TYPE | None:
+        """Subscribe to MQTT for a given event."""
+        data = self.coordinator.data
+        if (
+            event is None
+            or not mqtt.mqtt_config_entry_enabled(self.hass)
+            or not data["settings"]["mqttEnabled"]
+        ):
+            return None
+
+        @callback
+        def message_callback(message: mqtt.ReceiveMessage) -> None:
+            payload = json.loads(message.payload)
+            event_callback(**payload)
+
+        topic_template = data["settings"]["mqttEventTopic"]
+        topic = (
+            topic_template.replace("$appId", "fully")
```

Comment:
Please move all the MQTT data and topic parsing to the 3rd party library. The library can accept callbacks for MQTT subscribe and publish that Home Assistant can provide and then the library can handle the logic and parsing as needed.